index.bs

```diff
@@ -569,13 +582,21 @@ When this method is invoked, the user agent MUST execute the following algorithm
 
 1. Let |issuedRequests| and |currentlyAvailableAuthenticators| be new [=ordered sets=].
 
-1. For each |authenticator| currently available on this platform, if
-    <code>|options|.{{MakeCredentialOptions/attachment}}</code> is [=present|not present=] or its value matches
-    |authenticator|'s [=attachment modality=], [=set/append=] |authenticator| to |currentlyAvailableAuthenticators|.
-
-1. If |currentlyAvailableAuthenticators| [=list/is empty=], return a {{DOMException}} whose name is
+1. If |currentlyAvailableAuthenticators| [=list/is empty=], return a {{DOMException}} whose name is 
     "{{NotFoundError}}", and terminate this algorithm.
 
+1. If |options|.{{MakeCredentialOptions/authenticatorSelection}} is [=present|present=], iterate through 
+        |currentlyAvailableAuthenticators| and doing the following for each [=authenticator=]:
+    1. If |attachment| is [=present|not present=] or its value matches |authenticator|'s attachment modality, continue
+        to the next step. Otherwise, [=iteration/continue=] to the next [=authenticator=].
+    1. If |requireResidentKey| is set to |true| and the |authenticator| is capable of storing 
+        [=Client-Side-Resident Credential Private Key=], continue to the next step. Otherwise, return a {{DOMException}} 
+        whose name is "{{QuotaExceededError}}" and [=iteration/continue=] to the next [=authenticator=]. 
```

Comment:
This algorithm ultimately controls how [`navigator.credentials.create()`](https://w3c.github.io/webappsec-credential-management/#credentialscontainer) resolves the `Promise` it returned, and that `Promise` either rejects with an exception or fulfils with a `PublicKeyCredential`. There's no mechanism specified to return a stream of exceptions in addition to a `PublicKeyCredential`.

I think the `QuotaExceededError` solves one problem that @leshi is worried about, so it's worth keeping.

How about, add 
> 17. Let |selectionReasons| be a new list.

above this loop; change this step to:
> 2. If |requireResidentKey| is set to |true| and the |authenticator| is not capable of storing a [=Client-Side-Resident Credential Private Key=], append a new "{{QuotaExceededError}}" {{DOMException}} to |selectionReasons| and [=continue=].

; then change the step after this loop to:

> 19. If |selectedAuthenticators| is [=empty=], then if |selectionReasons| is:
    <dl class="switch">
    : [=empty=]
    :: Return a DOMException whose name is "ConstraintError", and terminate this algorithm.
    : [=not empty=]
    :: Return an arbitrary item of |selectionReasons|, and terminate this algorithm.
    </dl>

(I haven't double-checked that markdown; please make sure it renders sensibly.)